File: server/src/Cabinizer/Data/Booking.cs
using System;
using Microsoft.EntityFrameworkCore.Metadata.Builders;
using NodaTime;

namespace Cabinizer.Data
{
    public class Booking : Entity<Guid>
    {
        public string Name { get; set; } = null!;

        public Guid ItemId { get; set; }

        public Item Item { get; set; } = null!;

        public Guid RaffleId { get; set; }

        public Raffle Raffle { get; set; }

        public LocalDateTime StartTime { get; set; }

        public LocalDateTime EndTime { get; set; }

        public class Configuration : Configuration<Booking>
        {
            protected override void Configure(EntityTypeBuilder<Booking> builder)
            {
                builder.HasOne(x => x.Item)
                    .WithMany(x => x.Bookings)
                    .HasForeignKey(x => x.ItemId)
                    .IsRequired();

                builder.HasOne(x => x.Raffle)
                    .WithMany(x => x.Bookings)
                    .HasForeignKey(x => x.RaffleId)
                    .IsRequired();
            }
        }
    }
}

File: client-ts/src/api/cabinizer/models/Booking.ts
/* eslint-disable */

import { Instant } from './Instant';
import { Item } from './Item';
import { LocalDateTime } from './LocalDateTime';
import { Raffle } from './Raffle';

export interface Booking {
  id?: string;
  createdAt?: Instant;
  updatedAt?: Instant;
  createdBy?: string | null;
  updatedBy?: string | null;
  name?: string | null;
  itemId?: string;
  item?: Item;
  raffleId?: string;
  raffle?: Raffle;
  startTime?: LocalDateTime;
  endTime?: LocalDateTime;
}
